### ISSUE TITLE
Electrum + Ledger nano S BTC on Windows 10

### ISSUE THREAD

**Issue by dabeulioupee188** [spesmilo/electrum#6866]
When I try to connect my Ledger nano S (latest firmware version) using the BTC ledger app (version 1.5.1) with Electrum 4.0.8 and access the 2-of-4 multisignature wallet, I get this error message:
"Electrum cannot pair with your Ledger.

Before you request bitcoins to be sent to addresses in this wallet, ensure you can pair with your device, or that you have its seed (and passphrase, if any).  Otherwise all bitcoins you receive will be unspendable."

I tried: to connect with another USB cable, to connect using another PC, a MAC, to delete and reinstall Electrum, to use another version of Electrum (as it used to work in the past), to restart my computer, to turn off my VPN and Windows Defender. It did not help.
What should I do? It manages to connect when I am using an ETH multisig with Ledger or Electrum with Ledger in unisignature, but I cannot access my multisignature....

**Comment by scottdchris** [comment]
Same on:
- Electrum (4.0.9)
- Ledger Nano S (latest firmware) + BTC Ledger app (1.5.1)
- macOS Big Sur (11.1)
2-of-3 multisig wallet

My Ledger Nano X works perfectly fine but both Nano S have this issue. Sometimes Nano S is fine if I sign with Nano X first, but having trouble reproducing reliably.

**Comment by dabeulioupee188** [comment]
I spent some time with the Ledger CS and it might be that the derivation path chosen by default on the electrum software is wrong, and it cannot be changed..

**Comment by SomberNight** [comment]
Could you run the following line in the Qt console (`Console` tab in the GUI), and share its output?
```
[(ks.dump().get('type'), ks.dump().get('hw_type'), ks.dump().get('derivation'), ks.dump().get('root_fingerprint'), ks.dump().get('soft_device_id'), ks.dump().get('label')) for ks in wallet.get_keystores()]
```

e.g.
```
>>> [(ks.dump().get('type'), ks.dump().get('hw_type'), ks.dump().get('derivation'), ks.dump().get('root_fingerprint'), ks.dump().get('soft_device_id'), ks.dump().get('label')) for ks in wallet.get_keystores()]
[('hardware', 'coldcard', "m/48'/1'/0'/1'", 'b2768d2f', None, 'Coldcard b2768d2f'), ('hardware', 'trezor', "m/48'/1'/0'/1'", '34652fae', 'A3A1262C197E4C2E0855E090', 'trezorTreal')]
```

**Comment by dabeulioupee188** [comment]
[('hardware', 'ledger', "m/48'/0'/0'/2'", None, None, ''), ('hardware', 'ledger', "m/48'/0'/0'/2'", None, None, ''), ('hardware', 'ledger', "m/48'/0'/0'/2'", None, None, ''), ('hardware', 'ledger', "m/48'/0'/0'/2'", None, None, '')]

**Comment by SomberNight** [comment]
related: https://github.com/spesmilo/electrum/issues/5789

**Comment by SomberNight** [comment]
@dabeulioupee188 Electrum cannot figure out which keystore to pair your device with as there is no identifying info stored in your wallet file for some reason (compare my example output with yours)
Electrum normally iterates through the keystores in order, and tries to pair connected devices with them -- seeing what is in your wallet file, I expect if there are multiple devices connected, it will ask you to choose which one to use; if there is only one device connected, it will try to pair with that. If it tries to pair with the incorrect device, that's when you get the "cannot pair" error msg.

I am not sure how you managed to create that file however :/

Note that if any keystore manages to get paired successfully, then the missing identifying info should get persisted in the wallet file. So I am also confused about you saying "it used to work in the past" -- if it ever paired successfully some info should be there...

@scottdchris could you also run the command and share its output?

**Comment by scottdchris** [comment]
Thanks for investigating @SomberNight.

Here's my output:
```[('hardware', 'ledger', "m/48'/0'/0'/2'", '1b0eb9da', '1b0eb9da', ' '), ('hardware', 'ledger', "m/48'/0'/0'/2'", '69bb67a9', '69bb67a9', ' '), ('hardware', 'ledger', "m/48'/0'/0'/2'", '16c20bd4', '16c20bd4', ' ')]```

> Electrum normally iterates through the keystores in order, and tries to pair connected devices with them
> ...
> If it tries to pair with the incorrect device, that's when you get the "cannot pair" error msg.

If I'm understanding this correctly and order matters for some reason, maybe this is why I have intermittent success when I attempt to sign with my Nano X first. I think that was the first device I used when I created the multisig wallet.

**Comment by SomberNight** [comment]
@scottdchris 
> Here's my output:
`[('hardware', 'ledger', "m/48'/0'/0'/2'", '1b0eb9da', '1b0eb9da', ' '), ('hardware', 'ledger', "m/48'/0'/0'/2'", '69bb67a9', '69bb67a9', ' '), ('hardware', 'ledger', "m/48'/0'/0'/2'", '16c20bd4', '16c20bd4', ' ')]`

Hmmm. That looks perfectly fine. I would not expect any pairing issues :/ In any case the two of you probably have only tangentially related problems.

> My Nano X works perfectly fine but both Nano S have this issue. Sometimes Nano S is fine if I sign with Nano X first, but having trouble reproducing that reliably and unable to sign with the two Nano S only.

I have an idea. Could you test whether the differentiating factor is if the devices are connected at the moment the wallet file is opened. That is, can you test if you can reliably sign a tx/etc if the two devices you are going to use are already simultaneously connected the by the time the wallet file is opened?

**Comment by scottdchris** [comment]
@SomberNight 
Connecting devices before/after opening the wallet does seem to have an effect. Here are notes from some tests I did:

Nano X works whether connected before or after wallet is opened and alongside all other devices, so basically in all cases.

Both Nano S have the following symptoms (connected before wallet open):
- If 1 NX and 2 NS are connected: transaction can be confirmed on NX then electrum fails to connect to any NS.
- If 1 NX and 1 NS are connected: transaction can be confirmed on NX then electrum fails to connect to NS.
- If 0 NX and 1 NS is connected: transaction can be confirmed on NS1. Trying to connect NS2 without first closing the app will result in a failure.
- If 0 NX and 2 NS are connected: transaction can be confirmed on NS1 then failure to connect to NS2. If we disconnect NS1 then attempt to sign again, this dialog is shown before eventual failure: 
<img width="353" alt="Screen Shot 2020-12-23 at 7 02 57 PM" src="https://user-images.githubusercontent.com/5044720/103056965-54aa9680-4553-11eb-8a81-55c3b85fa1bc.png">

In conclusion I _do_ have at least one path that works if I only have my 2 NS:
1. Connect NS1
2. Open wallet
3. Sign transaction with NS1
4. Save partially signed transaction
5. Close wallet and disconnect NS1 (any order)
7. Connect NS2
8. Open wallet
9. Sign saved transaction with NS2
10. Broadcast

I'm satisfied with having at least one instruction set that works, but ideally the Nano S would play harmoniously with other devices the way Nano X does.

**Comment by dabeulioupee188** [comment]
> I am not sure how you managed to create that file however :/
> 
> Note that if any keystore manages to get paired successfully, then the missing identifying info should get persisted in the wallet file. So I am also confused about you saying "it used to work in the past" -- if it ever paired successfully some info should be there...
> 
---
Thanks for your quick reply. I am confused too as it did work in the past, when I was using older versions of Electrum and Ledger firmware. However now, even when using the past versions, it does not recognize the Ledger...
